src/parser.ts

```diff
@@ -207,6 +210,25 @@ export class Parser {
         this.jobs.forEach((job) => {
             job.producers = Producers.init(this.jobs, this.stages, job);
         });
+
+        if (this.argv.enableJsonSchemaValidation) {
+            const ajv = new Ajv({
+                verbose: true,
+                allErrors: true,
+                allowUnionTypes: true,
+                strictTypes: false, // to suppress the missing types defined in the gitlab-ci json schema
+            });
+            ajv.addKeyword("markdownDescription");
+            addFormats(ajv);
+            const validate = ajv.compile(schema);
+            const valid = validate(gitlabData);
+            assert(valid,
+                chalk`
+Invalid gitlab-ci configuration! It have failed the json schema validation. Dump the following to the pipeline editor to debug:
+${yaml.dump(gitlabData)}
```

Comment:
currently only dumping out the yaml... 

can't really find a nice way to parse the validate.errors that catches all scenarios

Reply:
I'm pretty sure I saw a way to pretty print AJV errors somewhere, let me dig around.